demo/src/main/java/com/novoda/movies/rateable/RateableMovieViewHolder.java

```diff
@@ -0,0 +1,120 @@
+        titleTextView = itemView.findViewById(R.id.item_rateable_text_title);
+        likeImageView = itemView.findViewById(R.id.item_rateable_image_like);
+        ratingBar = itemView.findViewById(R.id.item_rateable_rating);
+        a11yServices = AccessibilityServices.newInstance(itemView.getContext());
+    }
+
+    void bind(RateableMovieViewModel viewModel) {
+        posterImageView.setImageResource(viewModel.poster);
+        titleTextView.setText(viewModel.title);
+        likeImageView.setImageResource(viewModel.liked ? R.drawable.ic_favorite_24dp : R.drawable.ic_favorite_border_24dp);
+        ratingBar.setRating(viewModel.rating);
+        Actions actions = collateActionsFor(viewModel);
+        ActionsAccessibilityDelegate a11yDelegate = new ActionsAccessibilityDelegate(itemView.getResources(), actions);
+        ViewCompat.setAccessibilityDelegate(itemView, a11yDelegate);
+        itemView.setContentDescription(viewModel.title + ", rating " + viewModel.rating + ", liked: " + viewModel.liked);
+
+        if (a11yServices.isSpokenFeedbackEnabled() || a11yServices.isSwitchAccessEnabled() || !itemView.isInTouchMode()) {
+            bindForIndirectAccess(actions);
+            a11yDelegate.setClickLabel(R.string.action_rateable_movie_usage_hint_click);
+        } else {
+            bindForTouchAccess(viewModel);
+        }
+    }
+
+    private Actions collateActionsFor(RateableMovieViewModel viewModel) {
+        return new Actions(
+                Arrays.asList(
+                        selectActionFor(viewModel),
+                        toggleLikeActionFor(viewModel),
+                        rateActionFor(viewModel)
+                )
+        );
+    }
+
+    private void bindForIndirectAccess(Actions actions) {
+        itemView.setOnClickListener(v -> new ActionsAlertDialogCreator(itemView.getContext()).create(actions).show());
+        likeImageView.setClickable(false);
+        ratingBar.setOnRatingBarChangeListener(null);
+    }
+
+    private Action selectActionFor(RateableMovieViewModel viewModel) {
+        return new Action(R.id.action_rateable_movie_click_select, R.string.action_rateable_movie_click_select, () -> viewModel.actions.onSelectMovie());
+    }
+
+    private Action toggleLikeActionFor(RateableMovieViewModel viewModel) {
+        Runnable onToggleLikeRunnable = () -> viewModel.actions.onToggleLike();
+        if (viewModel.liked) {
+            return new Action(R.id.action_rateable_movie_click_toggle_like, R.string.action_rateable_movie_click_remove_like, onToggleLikeRunnable);
+        } else {
+            return new Action(R.id.action_rateable_movie_click_toggle_like, R.string.action_rateable_movie_click_like, onToggleLikeRunnable);
+        }
+    }
+
+    private Action rateActionFor(RateableMovieViewModel viewModel) {
+        return new Action(R.id.action_rateable_movie_click_rate, R.string.action_rateable_movie_click_rate, () -> showRatingDialog(viewModel));
```

Comment:
the rate action 😅 I tested it with switch access/talkback, wondering what the default behaviour is. I forgot to take screenshots, but essentially you have to select it and click "scroll forward/backward" one at a time to increment by the step size - which is 0.5* in this case. This is pretty bad on its own. But if you also consider that the focus is reset to the top of the activity each time you increment it's super bad.

For this reason, I decided to enumerate the ratings, and present it in _another_ list dialog. linear lists with a single click action in general are predictable to navigate and there are shortcuts to get through them on certain services.